packages/js/src/novu.test.ts

```diff
@@ -1,107 +1,78 @@
-import { ListNotificationsArgs } from './notifications';
 import { Novu } from './novu';
-import { NovuError } from './utils/errors';
+
+const mockSessionResponse = { data: { token: 'cafebabe' } };
 
 const mockNotificationsResponse = {
   data: [],
   hasMore: true,
   filter: { tags: [], read: false, archived: false },
 };
 
-const post = jest.fn().mockResolvedValue({ token: 'token', profile: 'profile' });
-const getFullResponse = jest.fn(() => mockNotificationsResponse);
-const updateHeaders = jest.fn();
-const setAuthorizationToken = jest.fn();
-
-jest.mock('@novu/client', () => ({
-  ...jest.requireActual('@novu/client'),
-  HttpClient: jest.fn().mockImplementation(() => {
-    const httpClient = {
-      post,
-      getFullResponse,
-      updateHeaders,
-      setAuthorizationToken,
+async function mockFetch(url: string, reqInit: Request) {
+  if (url.includes('/session')) {
+    return {
+      ok: true,
+      status: 200,
+      json: async () => mockSessionResponse,
     };
+  }
+  if (url.includes('/notifications')) {
+    return {
+      ok: true,
+      status: 200,
+      json: async () => mockNotificationsResponse,
+    };
+  }
+  throw new Error(`Unmocked request: ${url}`);
+}
 
-    return httpClient;
-  }),
-}));
+beforeAll(() => jest.spyOn(global, 'fetch'));
+afterAll(() => jest.restoreAllMocks());
 
 describe('Novu', () => {
+  const applicationIdentifier = 'foo';
+  const subscriberId = 'bar';
+
   beforeEach(() => {
-    jest.clearAllMocks();
+    // @ts-ignore
+    global.fetch.mockImplementation(mockFetch) as jest.Mock;
   });
 
-  describe('lazy session initialization', () => {
-    test('should call the queued notifications.list after the session is initialized', async () => {
+  describe('http client', () => {
```

Comment:
can you please add your tests separately? because the previous ones were covering lazy session initialization and queueing